packages/meteor/fiber_helpers.js

```diff
@@ -1,20 +1,5 @@
-var Fiber = Npm.require('fibers');
-var Future = Npm.require('fibers/future');
-
 Meteor._noYieldsAllowed = function (f) {
-  if (!Meteor._isFibersEnabled) {
-    return f();
-  }
-
-  var savedYield = Fiber.yield;
-  Fiber.yield = function () {
-    throw new Error("Can't call yield in a noYieldsAllowed block!");
-  };
-  try {
-    return f();
-  } finally {
-    Fiber.yield = savedYield;
-  }
+  return f();
```

Comment:
The purpose of noYieldsAllowed was to ensure that `f` did not have any async code. This type of check is still needed in some places, so it should be replaced with a version that does the same, but for promises.

Reply:
I did change this function to check this [cases](https://github.com/meteor/meteor/blob/b15dcc539376a68424f44de461d0d8b57e370436/packages/meteor/async_helpers.js#L1-L7).